autrainer/datasets/msppodcast.py

```diff
@@ -0,0 +1,202 @@
+            stratify=stratify,
+        )
+        self.standardize = standardize
+        if self.standardize:
+            train_data = torch.cat([x for x, *_ in self.train_dataset])
+            standardizer = Standardizer(
+                mean=train_data.mean(0).tolist(),
+                std=train_data.std(0).tolist(),
+            )
+
+            self.train_transform += standardizer
+            self.dev_transform += standardizer
+            self.test_transform += standardizer
+
+            self.train_dataset = self._init_dataset(
+                self.df_train, self.train_transform
+            )
+
+    @staticmethod
+    def download(path: str) -> None:  # pragma: no cover
+        """
+        Download the MSP-Podcast dataset.
+
+        As this dataset is not publicly-available, please download it manually
+        by contacting Prof. Carlos Busso: 
+        https://ecs.utdallas.edu/research/researchlabs/msp-lab/MSP-Podcast.html
+
+        This function will not do anything.
+
+        For more information on the data, see:
+        https://doi.org/10.1109/TAFFC.2017.2736999
+        """
+        return None
+    
+    def load_dataframes(
+        self,
+    ) -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+        """Load the dataframes.
+
+        Overrides base class so there is no need
+        to create separate train/dev/test files.
+
+        Returns:
+            Dataframes for training, development, and testing.
+        """
+        df = pd.read_csv(os.path.join(self.path, "Labels", "labels_consensus.csv"))
+        if self.categories is not None:
+            df = df.loc[df["EmoClass"].isin(self.categories)]
+
+        df_train = df.loc[df["Split_Set"] == "Train"]
+        df_dev = df.loc[df["Split_Set"] == "Development"]
+        df_test = df.loc[df["Split_Set"] == "Test1"]
+        return df_train, df_dev, df_test
+    
+    @cached_property
+    def target_transform(self) -> AbstractTargetTransform:
+        """Get the target transform.
+
+        Determined automatically based on the type of task.
+
+        Returns:
+            Target transform.
+        """
+        if self.task == "classification":
+            return LabelEncoder(
+                self.df_train[self.target_column].unique().tolist()
+            )
+        elif self.task == "regression":
+            return MinMaxScaler(
+                minimum=self.df_train[self.target_column].min(),
+                maximum=self.df_train[self.target_column].max(),
+            )
+        else:
+            raise NotImplementedError(f"{self.task} not supported for MSPPodcast")
+
+    @cached_property
+    def output_dim(self) -> int:
+        """Get the output dimension of the dataset.
+
+        Determined automatically based on the type of task.
+
+        Returns:
+            Number of classes.
+        """
+        if self.task == "classification":
+            return len(self.df_train[self.target_column].unique())
```

Comment:
Idea (not for here, just general): It may be a good idea to implement `__len__` for all target transforms, this could be more intuitive, do you agree?

Reply:
Yes, absolutely. Please open a new issue to track and resolve this

Reply:
Ref #49

Reply:
`output_dim` can now be safely removed and is handled by the `AbstractDataset` automatically.